### ISSUE TITLE
invoke-super

### ISSUE THREAD

**Issue by etosch** [jjfiv/dalvik-js#96]
Report any issues with `invoke-super` here. `invoke` code will likely move soon, so just make sure you push/pull and post any updates, problems etc. You mentioned having an issue with `invoke-super`. If you could post a detailed description of the problem, someone might be able to help.

**Comment by etosch** [comment]
@jen4ik, what's the status on `invoke-super`?

**Comment by jen4ik** [comment]
The status is that I spend time looking for changes made to code that I'm working on because people who modify it don't notify upon significant changes such as moving the code to a new file!!!

**Comment by etosch** [comment]
That's not a status; could you post what the specific problems are (e.g.
errors, state of the interpreter, etc), what test case you're using (with a
link), and the estimated time to fix the problem, along with your
availability?

On Tue, Oct 30, 2012 at 11:04 PM, jen4ik notifications@github.com wrote:

> The status is that I spend time looking for changes made to code that I'm
> working on because people who modify it don't notify upon significant
> changes such as moving the code to a new file!!!
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/jjfiv/dalvik-js/issues/96#issuecomment-9931255.

## 

Emma Tosch
Graduate Research Assistant, Computational Intelligence Lab, Hampshire
College
MS/PhD student, University of Massachusetts, Amherst
etosch@cs.umass.edu
http://cs.umass.edu/~etosch

**Comment by jen4ik** [comment]
That is the statement of my specific problem

**Comment by etosch** [comment]
The change was dicussed in #91, among other places. Always remember to stage and commit your work in the event of a manual merge, push code when you think you'll be away from it for a while, and pull before starting anew.

**Comment by etosch** [comment]
ah yes, but what is the problem with the output and what are the
impediments to `invoke-super` 's implementation? This would be of greater
interest to the group.

On Tue, Oct 30, 2012 at 11:11 PM, jen4ik notifications@github.com wrote:

> That is the statement of my specific problem
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/jjfiv/dalvik-js/issues/96#issuecomment-9931363.

## 

Emma Tosch
Graduate Research Assistant, Computational Intelligence Lab, Hampshire
College
MS/PhD student, University of Massachusetts, Amherst
etosch@cs.umass.edu
http://cs.umass.edu/~etosch

**Comment by jen4ik** [comment]
There is a difference between saying a change might be done, and then actually doing it and not notifying the other person that is working closely with the code.
I do always pull before I start working, but I cannot guess which changes I should be particularly looking for in such a broad project.

**Comment by etosch** [comment]
It's posted in the original commentary of this thread. I'll put it in _bold_ next time.  Please post an actual update however; it can even be something as simple as the following:

"I thought I had `invoke-super` working, but when I ran <filename>.dex in <webpage>.html, it threw a <name of error>'

If you observe anything about the state of the interpreter that's worth noting, please include that as well.

Finally per your request, @jen4ik, @jjfiv , @VJD215, @guptha - _Threads have been posted_. This is a _major change_ and definitely kills some tests. I am creating issues for known problems before I catch my bus home. Please post issues as soon as you come across them. 

@jjfiv - could you reassign tasks in light of these changes and expected delivery time for various features?

**Comment by etosch** [comment]
damn. stars make italics. /is this emboldened?/

Finally, as a wise fool, here's a word of advice: take initiative! It's grad school, after all. From the detritus of failure and trampled egos, the willful rise again.

**Comment by jen4ik** [comment]
I shall restate my initial comment - saying something might happen, which only happened a day after the initial issue appeared, doesn't not mean I actually know when it HAS indeed happened unless it's brought to our attention.

Regarding invoke-super - it is now working (i.e. no errors are thrown).
However, I am not certain that the correct registers are being passed to the invoked method, as the values printed as output don't match the original values in the java files.

Another issue I've noticed is that instance-get/put might not be working correctly - I get "undefined" printed instead of a string value

**Comment by jen4ik** [comment]
Availability - I have an assignment due on Thursday, should be able to spend most of my time on the VM after that

**Comment by jjfiv** [comment]
@jen4ik - if the issue you're talking about with instance-get/put is 
what we talked about this afternoon, put an issue in for it.

**Comment by etosch** [comment]
What is the test file for invoke-super?

On Tue, Oct 30, 2012 at 11:45 PM, John Foley notifications@github.comwrote:

> @jen4ik - if the issue you're talking about with instance-get/put is
> what we talked about this afternoon, put an issue in for it.
> 
> On 10/30/2012 11:42 PM, jen4ik wrote:
> 
> > I shall restate my initial comment - saying something might happen,
> > which only happened a day after the initial issue appeared, doesn't
> > not mean I actually know when it HAS indeed happened unless it's
> > brought to our attention.
> > 
> > Regarding invoke-super - it is now working (i.e. no errors are thrown).
> > However, I am not certain that the correct registers are being passed
> > to the invoked method, as the values printed as output don't match the
> > original values in the java files.
> > 
> > Another issue I've noticed is that instance-get/put might not be
> > working correctly - I get "undefined" printed instead of a string value
> > 
> > —
> > Reply to this email directly or view it on GitHub
> > https://github.com/jjfiv/dalvik-js/issues/96#issuecomment-9931935.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/jjfiv/dalvik-js/issues/96#issuecomment-9931961.

## 

Emma Tosch
Graduate Research Assistant, Computational Intelligence Lab, Hampshire
College
MS/PhD student, University of Massachusetts, Amherst
etosch@cs.umass.edu
http://cs.umass.edu/~etosch

**Comment by jen4ik** [comment]
@etosch - Bird.dex

**Comment by jen4ik** [comment]
@jjfiv I think there are 2 issues here
1. It is the one we talked about earlier today (I'll open an issue)
2. When we first talked about where should I put the invoke-super code, you suggested putting it before the code Emma wrote for register handling in invoke. I'm not sure that it's working the right way, but I need to investigate further.. The upside - we can use programs with calls to invoke-super, as long as we don't care about 100% accuracy in results at the moment. The instructions are being executed

**Comment by jjfiv** [comment]
Based on 2., have you checked in your code? It's entirely possible that 
all the issues you're seeing are from instance-get.

If you have questions about where to put invoke code, I'll have to 
direct you to @etosch, because I haven't touched it in a long time.

**Comment by jen4ik** [comment]
Yes, the code has been pushed, sorry, forgot to mention

It's not exactly a matter of where putting the code, but what is done with the registers that are supposed to be used (and / or how to override using the registers specified in the method call at the super class)
Could this mean re-parsing the method call so we can replace registers? (very not wanted)

**Comment by jjfiv** [comment]
If code has been pushed, is this issue complete? Or is there still an open issue about registers? (Blocked by the instance-get issue? #103?)

**Comment by jen4ik** [comment]
IMO, I would not call this happy as long as I cannot make sure that it at least calls the correct registers.
Right now, I cannot be sure whether it's the correct set of registers but they contain garbage, or it's the wrong set of registers
